lib/build.js

```diff
@@ -226,7 +227,7 @@ class BuildModel extends BaseModel {
                         isPR: job.isPR(),
                         jobId: job.id,
                         pipelineId: pipeline.id
-                    }, pipeline.scmContext) }))
+                    }, pipeline.scmContext, TEMPORAL_JWT_TIMEOUT) }))
```

Comment:
this sets an upper limit on how long a build can be in queue. I worry 30 mins won't be high enough for following use cases.

1. When we have [freeze windows](https://github.com/screwdriver-cd/screwdriver/issues/675).
2. Blocked_by jobs. Say a deploy job is waiting on a test job.

`TEMPORAL_JWT_TIMEOUT` value should be high enough to address this.

Reply:
I see. It should be long enough. The scope of this JWT is just limited, so It can be more long.
How long is appropriate?

Reply:
@sakka2 I would go for the `MAX_EXPIRES_IN` https://github.com/screwdriver-cd/screwdriver/pull/1087/files#diff-cc45542b37e76ced3e366e4aa20f512cR19 We can revisit when we get to `freeze windows`

That brings another question, rather than redefining the constant here, can we *move* the timeout constants from api to here in models such that everything is defined in a central place.

Reply:
@jithin1987 
I deleted the strategy of `maxAge` for JWT in [API here](https://github.com/screwdriver-cd/screwdriver/pull/1087/files#diff-cc45542b37e76ced3e366e4aa20f512cL129).
This is the optional verification that if expires time of JWT is over this value, it will be error.
So, if we can always set appropriate expires time to JWT, this verification will not be needed.

So, now we can declare the timeout constants only in this model.
I set `TEMPORAL_JWT_TIMEOUT` to `12h` as a default. I think it will be long enough for `Blocked by` jobs.

And then, when we get `freeze windows`, may be freezing time will be obtained from the annotation. even if it is 2days of 3days, we can set these terms to the JWT timeout.